### PR TITLE
Adopt Java 11 jdk.internal.reflect.Reflection.getCallerClass API

### PR DESCRIPTION
**Pull request by JasonFengJ9**
Adopt `Java 11 jdk.internal.reflect.Reflection.getCallerClass` API

Added `Java 11` specific libraries;
Invoke `JVM_GetCallerClass(env)` for `Java 11` which equivalent to
`JVM_GetCallerClass(env, -1)` for other Java versions.

Raw build `-version` output is as following:
```
openjdk version "11-internal" 2018-03-20
OpenJDK Runtime Environment (build 11-internal+0-adhoc.arnold.jdk111)
IBM J9 VM (build 2.9, JRE 11 Linux amd64-64 Compressed References 20180223_379402 (JIT enabled, AOT enabled)
OpenJ9   - 37a1735
OMR      - 58194bc
IBM      - 975f93d)
```

Reviewer @pshipton 
FYI: @DanHeidinga 

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

**Comment by pshipton** [comment]
Note the copyright failure

**Comment by pshipton** [comment]
Please fix the PR and commit title to describe the change.

**Comment by JasonFengJ9** [comment]
Updated PR accordingly, pls have another look.

File: runtime/sunvmi/sunvmi.c [hunk 1]
```diff
@@ -263,10 +263,13 @@ JVM_GetCallerClass_Impl(JNIEnv *env, jint depth)
 	
 	walkState.frameWalkFunction = getCallerClassIterator;
 
-	if (J2SE_18 <= (J2SE_VERSION(vm) & J2SE_VERSION_MASK)) {
-		if (-1 == depth) {
+	if (J2SE_18 <= J2SE_VERSION(vm)) {
```
Comment (pshipton) [inline runtime/sunvmi/sunvmi.c]: Since OpenJ9 only supports Java 8 or later, do you want to remove this test completely?

Comment (JasonFengJ9) [reply runtime/sunvmi/sunvmi.c]: Sure, removed.
Will use another PR to get rid of other `J2SE_18` checks.

File: runtime/sunvmi/sunvmi.c [hunk 2]
```diff
@@ -263,16 +263,17 @@ JVM_GetCallerClass_Impl(JNIEnv *env, jint depth)
...
-			 * [1] Method requesting caller
-			 * [0] actual caller
-			 */
-			depth = 2;
-			walkState.frameWalkFunction = getCallerClassJEP176Iterator;
-		}
+	/* Java 11 removed API getCallerClass(int depth),
+	 * getCallerClass() is equivalent to getCallerClass(-1).
+	 */
+	if ((-1 == depth) || (J2SE_V11 <= J2SE_VERSION(vm))) {
```
Comment (pshipton) [inline runtime/sunvmi/sunvmi.c]: Is this really going to work on all platforms? This JVM_GetCallerClass_Impl() has an extra parameter for Java 11, and should be declared `JVM_GetCallerClass_Impl(JNIEnv *env);`. Although its more work, another libjvm.so should be created to accommodate Java 11 differences.

Comment (JasonFengJ9) [reply runtime/sunvmi/sunvmi.c]: The `JVM_` method invoked by `JCL` was `JVM_GetCallerClass(env, depth)` which invokes `JVM_GetCallerClass_Impl` eventually.
Will look at `Java 11 libjvm.so` solution.

Comment (pshipton) [reply runtime/sunvmi/sunvmi.c]: If JVM_GetCallerClass(env, depth) is still being called on Java 11, what is the depth parameter set to?

**Comment by JasonFengJ9** [comment]
Updated the PR to build `Java 11` libraries to accommodate this API changes. 
Pls review again.

P.S. attempted to combine `sunvmi` (along with `clear_vmi`) with `j9vm` in hope to avoid making `Java 11 sunvmi` static library.  However `sunvmi` is needed by `jclse` as well though only `SunVMI_LifecycleEvent` not `JVM_xxx`; separating `JVM_xxx` methods from `sunvmi` didn't succeed due to the references to `JVM_xxx` within `SunVMI_LifecycleEvent`. In addition, moving `JVM_xxx` methods from `j9vm` to a static libary didn't succeed to dependency of `j9vm` from `JCL libjvm`. It appears more changes are required, probably leave it in a separated PR instead.

**Comment by pshipton** [comment]
Please fix the copyright failure in runtime/buildtools.mk

**Comment by pshipton** [comment]
@dnakamura fyi this change for #1009

**Comment by pshipton** [comment]
jenkins test sanity plinux jdk9

**Comment by pshipton** [comment]
test sanity plinux jdk9 passed https://ci.eclipse.org/openj9/job/PullRequest-Sanity-JDK9-linux_ppc-64_cmprssptrs_le-OpenJ9/81/

**Comment by JasonFengJ9** [comment]
Fixed the copyright in `runtime/buildtools.mk`

**Comment by pshipton** [comment]
The remaining copyright failure is expected as the file doesn't support a copyright.

**Pull request merged**